src/Microsoft.TestPlatform.CrossPlatEngine/PostProcessing/ArtifactProcessingManager.cs

```diff
@@ -65,7 +65,14 @@ internal class ArtifactProcessingManager : IArtifactProcessingManager
         {
             _testSessionCorrelationId = testSessionCorrelationId;
             _processArtifactFolder = Path.Combine(_fileHelper.GetTempPath(), _testSessionCorrelationId);
-            _testSessionProcessArtifactFolder = Path.Combine(_processArtifactFolder, $"{Process.GetCurrentProcess().Id}_{Guid.NewGuid()}");
+#if NET5_0_OR_GREATER
+            var pid = Environment.ProcessId;
```

Comment:
is it cached?

Reply:
System.Diagnostics.Process.GetCurrentProcess().Id is expensive:

It allocates a [Process](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process) instance, usually just to get the Id.
The [Process](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process) instance needs to be disposed, which has a performance impact.
It's easy to forget to call [Dispose()](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process.dispose#system-diagnostics-process-dispose) on the [Process](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process) instance.
If nothing else besides Id uses the Process instance, then the linked size grows unnecessarily by increasing the graph of types referenced.
It is somewhat difficult to discover or find this API.
System.Environment.ProcessId avoids all the above.